sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+criteria must require the following:
+
+   * At least 80% of all miners must _explicitly and continuously vote to
+     accept_ the soft fork rules for at least _two consecutive_ whole PoX reawrd
+cycles. Additional, stricter criteria are permitted.
+   * An _activation window_, defined by a _start activation height_ and an _end
+     activation height_, represented as burnchain block heights, must be
+stipulated in the activation criteria.  The aforementioned miner vote must take
+place and run to completion in-between these two block heights.  The block
+heights _must_ correspond to the beginnings of PoX reward cycles, and must be
+between 14 reward cycles (1 maximum Stacking period plus two reward cycles) and
+25 reward cycles (about 1 year) apart.
+   * A soft-fork-specific bit pattern must be specificied, which miners _must_
+     include in their coinbase transactions in order to indicate unambiguous
+support for activating the new rules.
+
+3. Once a release with the soft fork voting rules and activation window is
+written and adequately tested (possibly including rolling it out to one or more
+public testnets first), it will be announced via a signed email from
+announce@stacks.org.  This release announcement must occur before the activation
+window begins.
+4. If at least 80% of all miners vote to accept the soft fork rules for two
+consecutive, whole reward cycles within the activation window, then the new
+rules will take effect starting in the next whole reward cycle.  All new
+releases of the Stacks node will adhere to the soft fork rules, since they are
+now part of the block validation rules.
+5. A signed email from announce@stacks.org will be sent once the new rules take
+effect, urging all network participants to upgrade to a node that applies the
+new rules.
+
+Note that in this procedure, at least three emails from announce@stacks.org will
+be sent -- one in step 1, one in step 3, and one in step 5.
+
+## Procedure for Hard Forks
+
+A hard fork is backwards-incompatible blockchain upgrade.  They are to be
+avoided if at all possible, because they are tantamount to making an entirely
+new version of the blockchain that, if executed poorly, threatens to split the
+ecosystem into two or more parts.  A hard fork is a fundamental re-working of
+the social contract that the Stacks blockchain represents, and a successful hard
+fork requires near-unanimous agreement on the new rules from all ecosystem
+participants.
+
+A hard fork is only necessary for repairing catastrophic failures if all other
+procedures prove inadequate.  It is not to be used in any other circumstance.
+Therefore, the kinds of catastrophic failures that necessitate hard forks are
+artifacts of a design bug in the Stacks blockchain that renders it unsustainable
+or unable to operate as intended.
+
+Some examples of when a hard fork is the appropriate course of action include,
+but are not limited to:
```

Comment:
I would also add something like "Introducing new features or capabilities that the ecosystem is overwhelming in favor of". For instance, improvements to Stacking would fall into this category.

Reply:
If the ecosystem is overwhelmingly in favor of doing something with the urgency you suggest, then a SIP is neither going to stop them nor give them permission.